### PR TITLE
feat(autofix): Fix UI5 Bootstrap Parameters in HTML

### PR DESCRIPTION
**Pull request by matz3**
JIRA: CPOUI5FOUNDATION-1063

Open tasks:
- [x] Removal of binding syntax attribute if `compat-version=edge` is currently handled in addition to the existing check. This should be streamlined and tested properly.
- [x] Removal of duplicate attributes (identical names) + tests
- [x] Remove Attributes: Adjust logic to remove entire line (no whitespace)
- [x] Check existing fixtures and snapshots for correctness
- [x] Identify missing scenarios and add test cases
  - Missing test case for removal of binding syntax attribute if `compat-version=edge`
  - Maybe also some more missing cases
- [x] Implement parsing error checks (currently commented out in `src/autofix/autofix.ts`). This needs some investigation to find a good solution:
  - ~~HTML is not strictly valid XML, so we can't reuse the approach for XML files~~
  - ~~A separate HTML parser might be required to check for issues~~
  - ~~We might use the existing XML parser approach and only check for error differences of the file before and after the autofix~~
  - ~~Unit test for error handling could be done in a similar way as for XML changes: https://github.com/UI5/linter/blob/bc47a764bce39a127e59947f07fa4b505abe74b8/test/lib/autofix/autofix.ts#L41~~
  - Discussion on July 28: Agreed upon to omit a special HTML syntax validation since it's too vague and not very suitable for our cases -> Instead try to validate by covering as many edge cases in the tests as possible
- [x] Check code coverage and add unit tests where applicable (e.g. for `AnimationModeFix.getAnimationModeValue`)
- [x] Add test cases for unquoted attribute values (e.g. `data-sap-ui-compatVersion=edge`). Modifications should not fail or create invalid HTML, but it would be also fine to ignore such cases as it is unusual and rarely appears in real world projects
  - Add tests to ensure autofix doesn't break anything for these cases
- [x] Add note to [README.md](https://github.com/UI5/linter/blob/0a91ba13d6774e50e3ef166e67b402ccf7bcde07/README.md#--fix)
- [x] Handle removal of duplicate attributes **WITHOUT** values
- [x] Handle attributes which only have one character (e.g. `a="xyz"` or `a`)
  - Removal
  - Renaming
- [x] Add unit tests for RemoveAttributeFix, RenameAttributeFix & AnimationModeFix
- [x] Check Linter Diff

### CODE REVIEW

File: src/linter/html/fix/RemoveAttributeFix.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		}
+		if (attr.type === AttributeType.NoValue && attr.name.value.length === 1) {
+			endPos = {
+				line: attr.name.start.line,
+				character: attr.name.start.character + 1,
+			};
+		}
+
+		// Edge case: no space w/ NoValues or NoQuotes
+		const edgeCaseStartPos = this._handleEdgeCase(tag, attrIndex);
```
[the remark this answers is not in the document]
Comment (maxreichmann) [reply src/linter/html/fix/RemoveAttributeFix.ts:103]: Having two different methods for calculating each position seems reasonable. Will do that!
The logic was pretty small compared to now and that's why it's inside one method only currently.

[the remark this answers is not in the document]
Comment (maxreichmann) [reply src/linter/html/fix/RemoveAttributeFix.ts]: Done in [9ecc237](https://github.com/UI5/linter/pull/779/commits/9ecc237e6a1e2ed31c88576b9bc7c14d4dbd355a)

File: src/linter/html/fix/RemoveAttributeFix.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			throw new Error("Start and end position are not defined");
+		}
+		return {
+			action: ChangeAction.DELETE,
+			start: this.startPos,
+			end: this.endPos,
+		};
+	}
+
+	_handleEdgeCase(tag: SaxTag, attrIndex: number): PositionDetail | undefined {
```
[the remark this answers is not in the document]
Comment (maxreichmann) [reply src/linter/html/fix/RemoveAttributeFix.ts]: Done in [9ecc237](https://github.com/UI5/linter/pull/779/commits/9ecc237e6a1e2ed31c88576b9bc7c14d4dbd355a)

File: src/linter/html/HtmlReporter.ts [hunk 3]
```diff
@@ -44,10 +45,19 @@ export default class HtmlReporter {
...
+		};
+
+		const message = {
+			id,
+			args,
+			position,
+			fix,
+		};
+
+		this.#context.addLintingMessages(this.#resourcePath, [message]);
```
Comment (matz3) [inline src/linter/html/HtmlReporter.ts:60]: Is there a reason to switch from `this.#context.addLintingMessage` to `this.#context.addLintingMessages`?

**Review by d3xter666 — APPROVED** [review]
LGTM

File: src/linter/html/HtmlReporter.ts (hunk 3 above)
Comment (matz3) [reply src/linter/html/HtmlReporter.ts]: Done

**Review by matz3 — COMMENTED** [review]
LGTM

**Review by maxreichmann — APPROVED** [review]
